tests/src/core/particle/ParticleCurve.test.ts

```
@@ -0,0 +1,74 @@
+    expect(compositeCurve.evaluate(1.0, 1.0)).to.equal(0.8);
+
+    expect(compositeCurve.evaluate(0.6, 0.5)).to.equal(0.6499999999999999);
+  });
+
+  it("Add and remove", () => {
+    const curve = new ParticleCurve(new CurveKey(0, 0.3), new CurveKey(0.6, 0.7));
+ +  expect(curve.keys.length).to.equal(2);
+ +  
+ 
+    curve.addKey(new CurveKey(0, 0.4));
+ +  expect(curve.keys.length).to.equal(3);
```

ESLint

[error] Replace `+··` with `···+`

(prettier/prettier)